generate/generate.go

```diff
@@ -289,102 +291,54 @@ func (g generatedFiles) add(filename string, genfile generatedFile) error {
 	return nil
 }
 
-func updateGenFileOutdatedFiles(
-	root, stackpath string,
-	stackMeta stack.Metadata,
-	globals stack.Globals,
+func updateOutdatedFiles(
+	stackpath string,
+	genfiles generatedFiles,
 	outdatedFiles *stringSet,
 ) error {
 	logger := log.With().
-		Str("action", "generate.updateGenFileOutdatedFiles()").
-		Str("root", root).
+		Str("action", "generate.updateOutdatedFiles()").
 		Str("stackpath", stackpath).
 		Logger()
 
-	logger.Trace().Msg("Checking for outdated generated_file code on stack.")
-
-	stackFiles, err := genfile.Load(root, stackMeta, globals)
-	if err != nil {
-		return err
-	}
-
-	logger.Trace().Msg("Loaded generated_file code, checking")
+	logger.Trace().Msg("Checking for outdated generated_hcl code on stack.")
 
-	for filename, genFile := range stackFiles.GeneratedFiles() {
+	for filename, genHCL := range genfiles {
 		targetpath := filepath.Join(stackpath, filename)
 		logger := logger.With().
 			Str("blockName", filename).
 			Str("targetpath", targetpath).
 			Logger()
 
-		logger.Trace().Msg("Checking if file content is updated.")
+		logger.Trace().Msg("Checking if code is updated.")
 
-		fileContents, found, err := readFile(targetpath)
+		currentHCLcode, codeFound, err := readFile(targetpath)
```

Comment:
same here :thinking:

Reply:
yeah sorry, I refactored it to be general but then messed up fixing the names, thanks for the catch :heart: